Feature | Added Todos Page For User

saramkhalil opened the pull request:
- Implemented Todo page that lists all todos belonging to the logged-in user
- Added ability for users to mark their todos as done (status update)
- Enabled editing of existing todos directly from the page
- Integrated with access control to ensure todos are user-specific
- Improved UI/UX for managing todos efficiently

saramkhalil committed:
feat(todo): add Todo page with user-specific todos

- Implemented Todo page that lists all todos belonging to the logged-in user
- Added ability for users to mark their todos as done (status update)
- Enabled editing of existing todos directly from the page
- Integrated with access control to ensure todos are user-specific
- Improved UI/UX for managing todos efficiently

Asad170170 commented on controllers/SiteController.php:
```diff
@@ -62,9 +65,90 @@ public function actions()
     {
-        return $this->render('index');
+        if (\yii::$app->user->isGuest) {
+            return $this->redirect(['site/login']);
```
You repeat this logic in multiple actions

Extract to a helper method or use AccessControl more extensively

Asad170170 commented on controllers/SiteController.php:
```diff
@@ -62,9 +65,90 @@ public function actions()
+            return $this->redirect(['site/login']);
+        }
+    
+        $todo = Todo::find()
```
Reuse Common Queries via Model Scopes

Asad170170 commented on models/Todo.php:
```diff
@@ -0,0 +1,43 @@
+        ];
+    }
+
+    public function attributeLables()
```
Typo: attributeLables() → attributeLabels()

Asad170170 commented on models/Todo.php:
```diff
@@ -0,0 +1,43 @@
+
+class Todo extends ActiveRecord
+{
+    const STATUS_PENDING = 0;
```
Add Status Labels (Helper Method)

For better code readability in views, add a method something likee:

```
public static function getStatusList()
{
    return [
        self::STATUS_PENDING => 'Pending',
        self::STATUS_DONE => 'Done',
    ];
}

public function getStatusLabel()
{
    return self::getStatusList()[$this->status] ?? 'Unknown';
}
```

Then in views access like this:

`echo $todo->getStatusLabel();`

saramkhalil replied:
Addressed: added getStatusList() and getStatusLabel() in models/Todo.php and used in views/site/index.php

Asad170170 commented on views/site/index.php:
```diff
@@ -1,53 +1,71 @@
+            </li>
+        <?php endforeach; ?>
+    </ul>
+<?php endif; ?>
```
Add one extra line at the end.

Asad170170 commented on controllers/SiteController.php:
```diff
@@ -62,9 +65,90 @@ public function actions()
+
+        if ($todo->load(\yii::$app->request->post())) {
+            $todo->user_id = \Yii::$app->user->id;
+            $todo->status = Todo::STATUS_PENDING;
+        
+
```
Please avoid excessive blank lines. If you need to improve readability, one blank line is enough no need for multiple.

saramkhalil committed:
feat(todo): add query scopes and status labels; centralize auth

- Add models/TodoQuery with scopes: forUser(), byId(), pending(), done(), recent()
- Override Todo::find() to return TodoQuery
- Use scopes in SiteController for index/toggle/update paths
- Add Todo::getStatusList() and getStatusLabel(); update view to use label
- Protect index/toggle-todo/update-todo via AccessControl; remove per-action guest checks
- Fix toggle/update issues (inverted guest check and  check)
- Minor view tidy: single flash renderer, item separators
